### PR TITLE
fix: recognize immutable fields in lists

### PR DESCRIPTION
**Pull request by JulissaDantes**
## Description
In the current version a field declared as `uniqueListValue: [Int] @list(maxLength: 5) @immutable` is not included in the ImmutableFields. This PR allows to identify these kinds of declarations.

### CODE REVIEW

**Review by stbrody — COMMENTED** [review]
are there any tests of actually trying to update a MID with a model using `@immutable`?  The test in this PR is really just unit testing the parser rather than the actual user-visible behavior

**Review by stbrody — COMMENTED** [review]
lgtm, though I still think it'd be good for Paul to take a look.

Also CI is broken

File: packages/runtime/test/runtime.test.ts [hunk 1]
```diff
@@ -1042,13 +1045,16 @@ describe('runtime', () => {
...
+      i: { id: docID, content: { title: 'A different title' }, testList: ['d', 'e'] },
     })
-
-    expect(res2.errors).toBeDefined()
+    
+    expect(res2.errors?.length).toBe(2)
     expect(res2.errors![0].message).toEqual(
       `Variable "$i" got invalid value { title: "A different title" } at "i.content"; Field "title" is not defined by type "PartialPostInput". Did you mean "date"?`,
     )
+    expect(res2.errors![1].message).toContain(
```
Comment (PaulLeCam) [inline packages/runtime/test/runtime.test.ts:1055]: I don't really like the idea of testing the immutable fields handling based on parsing GraphQL queries, this seems pretty fragile.
I think instead there should be a direct call to the Ceramic server trying to make an update to an immutable field and getting the matching error message, what do you think?

**Review by PaulLeCam — CHANGES_REQUESTED** [review]

**Comment by JulissaDantes** [comment]
Marking as draft until I fix the CI

**Comment by JulissaDantes** [comment]
Ci fixed here: #228

File: packages/devtools/src/schema/parser.ts [hunk 2]
```diff
@@ -288,7 +288,10 @@ export class SchemaParser {
         immutableFields: Array.from(
           new Set(
             Object.keys(object.properties)
-              .filter((key) => object.properties[key].immutable === true)
+              .filter((key) => {
+                const property = object.properties[key];
+                return property.immutable === true || ('item' in property && property.item && property.item.immutable)
```
[the remark this answers is not in the document]
Comment (JulissaDantes) [reply packages/devtools/src/schema/parser.ts]: You are right, change made!

File: packages/runtime/test/runtime.test.ts [hunk 3]
```diff
@@ -1042,13 +1045,16 @@ describe('runtime', () => {
     }`
 
     const res2 = await runtime.executeQuery(updatePost, {
-      i: { id: docID, content: { title: 'A different title' } },
+      i: { id: docID, content: { title: 'A different title' }, testList: ['d', 'e'] },
```
[the remark this answers is not in the document]
Comment (JulissaDantes) [reply packages/runtime/test/runtime.test.ts]: thanks for spotting this

File: packages/runtime/test/runtime.test.ts (hunk 1 above)
Comment (JulissaDantes) [reply packages/runtime/test/runtime.test.ts:1055]: Agreed. FYI I'll close this PR, rebasing from main didnt helped, but I created a new branch branch with the changes(missing this last change) and I'll ask for a review on the new one that doesnt seem to have the CI issue

**Pull request closed without merging**